src/components/login/SignUp.tsx

```diff
@@ -15,26 +16,58 @@ import { Flex } from '../primitives/Flex';
 import { useAuth } from '../../contexts/auth';
 import { theme } from '../../theme';
 import { ChangeLoginMode } from './ChangeLoginMode';
+import validator from 'validator';
 
 type SignUpProps = { changeMode: () => void };
 export const SignUp = ({ changeMode }: SignUpProps) => {
   const [name, setName] = useState('');
   const [email, setEmail] = useState('');
   const [password, setPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
   const [error, setError] = useState('');
   const [isLoading, setLoading] = useState(false);
   const { createAccount } = useAuth();
 
+  const toast = useToast();
+
+  const toastErrorTemplate = () => {
+    const id = 'uniqueToast';
+    if (!toast.isActive(id)) {
+      toast({
+        id,
+        title: 'Error',
+        description: error,
+        duration: 3000,
+        isClosable: true,
+        status: 'error',
+        position: 'top-right',
+      });
+    }
+  };
+
   const handleAuthenticate = async () => {
     setError('');
 
+    if (!validator.isEmail(email)) {
```

Comment:
i don't think we need that, my mistake. if we encapsulate all the inputs from the form in a InputGroup, Chakra will validate the form for us!

Reply:
can you explain better?